client/knowledge_base_search/src/containers/ResultsContainer/ResultComparison.js

```diff
@@ -32,49 +33,80 @@ class ResultComparison extends Component {
   }
 
   getFullAnswer() {
-    const { full_result_type, result, enriched_result } = this.props;
+    const { full_result_type, result, enriched_result, passages } = this.props;
 
-    return full_result_type === 'enriched' ? enriched_result : result;
+    switch(full_result_type) {
+      case 'enriched':
+        this.setState({full_answer: enriched_result});
+        break;
+      case 'passages':
+        Promise.all([
+          query('enriched', {query: "id: "+passages.document_id})
+        ]).then((doc_by_id) => {
+          this.setState({full_answer: doc_by_id[0].results[0]});
+        });
+        break;
+      default:
+        this.setState({full_answer: result});
```

Comment:
no: this section determines what data is being used for the full result---if it is the plain result or a passage result. If the full result is not of type 'passages' it will assume it is a plain full result. we can add more `case`s in the future when we are showing more than just passages as a comparison.